frontends/web/src/components/mobiledatawarning.tsx

```diff
@@ -15,27 +15,25 @@
  * limitations under the License.
  */
 
-import { PropsWithChildren } from 'react';
-import { subscribe } from '../decorators/subscribe';
-import { translate, TranslateProps } from '../decorators/translate';
+import { FunctionComponent } from 'react';
+import { useTranslation } from 'react-i18next';
+import { useLoad, useSubscribe } from '../hooks/api';
+import { getUsingMobileData, subscribeUsingMobileData } from '../api/mobiledata';
 import Status from './status/status';
 
-interface LoadedProps {
-    usingMobileData: boolean;
-}
-
-type Props = LoadedProps & TranslateProps;
-
-function MobileDataWarning({ usingMobileData, t }: PropsWithChildren<Props>) {
+export const MobileDataWarning: FunctionComponent = () => {
+    const { t } = useTranslation();
+    const isUsingMobileData = useLoad(getUsingMobileData);
```

Comment:
i dislike that before it was one subscribe hoc and now it is two separate hooks (`useLoad` and `useSubscribe`), two functions `getUsingMobileData` and `subscribeUsingMobileData`, and double the logic below (`usingMobileData === false || !isUsingMobileData`).

This PR contains proof of that this is bugprone :) `getUsingMobileData` is calling the wrong endpoint.

In the old subscribe HOC, the subscription name is always the same as the endpoint name. I suggest that you add a combined subscribe/load hook that calls apiGet on the endpoint on load, as well as updates the value by listening on the same name in the subscription, to reduce all the complexity described above. WDYT?

Reply:
well technically it called 2 decorator HOC's before, `subscribe` and `load`

> one subscribe hoc 

thanks for catching the wrong endpoint, will fix this. Do you want that the endpoint is a constant that is reused? maybe lets chat offline

Reply:
I'd wish if `useSubscribe` would take a 2nd argument to set the default, but it could not default to  `useLoad(getUsingMobileData)` because that is async too and will first be undefined.